src/peft/optimizers/riemannian.py

```diff
@@ -0,0 +1,294 @@
+import math
+from operator import attrgetter
+from typing import Callable, Iterable, Tuple
+
+import torch
+import torch.nn as nn
+from torch.optim import Optimizer
+from transformers.utils.versions import require_version
+
+from ..peft_model import PeftModel
+
+
+def create_riemannian_optimizer(
+    model: PeftModel,
+    optimizer_cls: type[Optimizer],
+    optimizer_kwargs: dict,
```

Comment:
Since you probably took this from the LoRA+ PR, let me refer to the [comment](https://github.com/huggingface/peft/pull/1915#discussion_r1680736766) I put there:

A suggestion: Let's remove `optimizer_kwargs` and just add `**kwargs`. IMO, that makes calling this function easier, as we can use `create_riemannian_optimizer(..., weight_decay=1e-3)` instead of `create_riemannian_optimizer(..., optimizer_kwargs={..., "weight_decay": 1e-3})`. And since `lr` is not optional, let's make this a normal arg of `create_riemannian_optimizer`.